Makefile

```diff
@@ -476,9 +475,10 @@ govet: ## Run govet on Go source files in the repository.
     ./test/k8sT/... \
     ./tools/...
 
-lint: ## Run golangci-lint.
+lint: ## Run golangci-lint and check if the helper headers in bpf/mock are up-to-date.
 	@$(ECHO_CHECK) golangci-lint
 	$(QUIET) golangci-lint run
+	$(QUIET) $(MAKE) $(SUBMAKEOPTS) -C bpf/mock/ check_helper_headers
```

Comment:
hm...

Seems to me that if someone typed `make lint`, they wouldn't expect to run something related to `bpf/mock`. Do we maybe need another type for this?

Reply:
I agree it's not the most obvious place, but at least this will get automatically run by the existing github actions.

Do you have a concrete proposal for somewhere else and how that should be hooked into the github actions?

(I don't disagree, but also I feel like we can always improve this iteratively if it's a problem for developers)

Reply:
^^ There was also some background earlier in the PR, I can't exactly find it but the idea is that when someone updates the bpf helpers, we need to update also the generated mock helpers files, so it's sort-of like a linter in that regard.

Reply:
`the idea is that when someone updates the bpf helpers, we need to update also the generated mock helpers files, so it's sort-of like a linter in that regard`

That's the idea. `check_helper_headers` will report an error if the helper headers are out-of-date, and developers should run `generate_helper_headers` to update them.

Reply:
> (I don't disagree, but also I feel like we can always improve this iteratively if it's a problem for developers)

OK, sounds good!